### ISSUE TITLE
trigger for the bell / binary_sensor.doorbell_g4_

### ISSUE THREAD

**Issue by RW-7** [briis/unifiprotect#382]
### Home Assistant Version?

2021.11.5

### UniFi Protect Version?

1.20.1

### UniFi Protect HA Integration version?

0.11.0-beta.1

### What happened?

the trigger for the doorbell is unfortunately no longer available, or there is no longer any reaction to the actuation of the doorbell button (binary_sensor.doorbell_g4_).
The configuration for template texts for the display has also disappeared in beta 11

**Comment by AngellusMortis** [comment]
Can you please update your version of the HA integration to the correct version? 1.18.0 is not a valid version.

I am not sure I follow what your issue is. Are you saying that your doorbell is not ringing? Or that HA is not picking up on ring events?

Also, the config option for the custom doorbell messages was removed because we are using the custom doorbell messages now that UniFi Protect adds. You can either use the UniFi Protect Android/iOS app to add/remove new options or use the `unifiprotect.add_doorbell_text` and `unifiprotect.remove_doorbell_text` services. Note: because of how downstream subscriptions work for selects, you will need to restart Home Assistant after adding/removing any options.

**Comment by RW-7** [comment]
version adapted.
if someone rings the doorbell, I cannot evaluate this HA. so far this went with:
`type: occupied
platform: device
device_id: 52c7d6dac18f82880a55fa4303535f41
entity_id: binary_sensor.doorbell_g4_doorbell_garage
domain: binary_sensor
`
How do I have to adapt this for the automation?

**Comment by AngellusMortis** [comment]
I recommend checking out the blueprints I made:

https://community.home-assistant.io/t/unifi-protect-blueprints/362604/

You can find the raw source for the Doorbell one at: 

https://github.com/briis/unifiprotect/blob/blueprints/blueprints/automation/unifiprotect/doorbell_event.yaml

My trigger condition is a bit more complex because I am supporting the fancy select HA provides for Area/Device/Entity, but basically it should boil down to:

```yaml
trigger:
  - platform: state
     entity_id: binary_sensor.doorbell_front_doorbell_camera
     from: 'off'
     to: 'on'
```

**Comment by RW-7** [comment]
have now adjusted this with the beta 2 it does not trigger anything

`alias: Dorbell Gasse
description: ''
trigger:
  - platform: state
    entity_id: binary_sensor.doorbell_g4_doorbell_gasse
    from: 'off'
    to: 'on'
condition:
  - condition: time
    after: '06:59'
    before: '22:00'
action:
  - service: sonos.snapshot
    data:
      entity_id: media_player.wohnzimmer
  - service: media_player.volume_set
    target:
      device_id:
        - 434ztrjfgdsz
    data:
      volume_level: 0.69
  - service: tts.google_say
    data:
      entity_id: media_player.wohnzimmer
      message: Es klingelt an der Gassentür
      language: de
  - delay:
      hours: 0
      minutes: 0
      seconds: 3
      milliseconds: 0
  - service: sonos.restore
    data:
      entity_id: media_player.wohnzimmer
mode: restart`

**Comment by AngellusMortis** [comment]
Do you have any logs or errors that could indicate an issue? Have you tried restarting HA after upgrading?

**Comment by AngellusMortis** [comment]
Can you enable debug logging? After you do, restart HA and then post a log? Feel free to DM it me on the Home Assistant forums (same username) if you do not want to post it publically.

https://github.com/briis/unifiprotect#enable-debug-logging

**Comment by RW-7** [comment]
Logger: homeassistant.components.binary_sensor
Source: custom_components/unifiprotect/binary_sensor.py:351
Integration: Binärsensor (documentation, issues)
First occurred: 15:27:26 (4 occurrences)
Last logged: 15:45:51

Error while setting up unifiprotect platform for binary_sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/unifiprotect/binary_sensor.py", line 178, in async_setup_entry
    entities: list[UnifiProtectBinarySensor] = _async_device_entities(entry_data)
  File "/config/custom_components/unifiprotect/binary_sensor.py", line 260, in _async_device_entities
    UnifiProtectBinarySensor(
  File "/config/custom_components/unifiprotect/binary_sensor.py", line 300, in __init__
    self._async_update_device_from_protect()
  File "/config/custom_components/unifiprotect/binary_sensor.py", line 351, in _async_update_device_from_protect
    self._attr_is_on = (now - last_ring) < RING_INTERVAL
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'

and 

Logger: homeassistant.config_entries
Source: helpers/entity_component.py:164
First occurred: 15:45:55 (8 occurrences)
Last logged: 15:45:55

Error setting up entry unifi for switch
Error setting up entry unifi for light
Error setting up entry unifi for select
Error setting up entry unifi for number
Error setting up entry unifi for media_player
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 304, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 362, in async_setup_entry
    return await component.async_setup_entry(entry)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 164, in async_setup_entry
    raise ValueError("Config entry has already been setup!")
ValueError: Config entry has already been setup!

yes of course i restarted.

it would be nice if you could include the sonos notification in your blueprint

**Comment by AngellusMortis** [comment]
> it would be nice if you could include the sonos notification in your blueprint

I thought about that, but I think I would rather do a separate blueprint for that. Rather then having a massive automation that does _everything_ have 1 that does push notifications, one that does sound notifications, one that does light, etc.

**Comment by RW-7** [comment]
true, but this is how you can configure a device at once

**Comment by AngellusMortis** [comment]
> TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'

That looks like the issue there. Your doorbell has _never_ been rung? That does not seem right, it has to be a reporting issue. As a workaround, try ringing it?

**Comment by RW-7** [comment]
with 0.10.x it works without any problems, is in the end the same senor as I have seen.
yes, the doorbell rings often, also for testing

**Comment by AngellusMortis** [comment]
I want to figure out what is different about your setup to prevent the issue in the future. So, a bit of debugging if you will bear with me:

* Get the device ID for your G4 Camera by opening the settings for your G4 Camera _in your Protect Web application_. Then in the URL, it will be bit at the end of the URL (example: `/protect/devices/61830576039e5403e7000401` -> id=61830576039e5403e7000401

* Go to `/proxy/protect/api/cameras/{id}` in the same Web browser. Replace `{id}` with the value from above. You should get a bit JSON response that looks something like this:

![image](https://user-images.githubusercontent.com/490848/145058027-608469ad-80ac-4445-b1fd-0a4e2124ec38.png)


Post that JSON back here for me if you could.

**Comment by RW-7** [comment]
`{"isDeleting":false,"mac":"68D79ACF5EE9","host":"192.168.4.250","connectionHost":"192.168.4.4","type":"UVC G4 Doorbell","name":"G4 Doorbell Garage ","upSince":1638868153533,"uptime":23107,"lastSeen":1638891260533,"connectedSince":1638868184513,"state":"CONNECTED","hardwareRevision":"22","firmwareVersion":"4.46.18","latestFirmwareVersion":"4.46.18","firmwareBuild":"ceacbaa.211202.1017","isUpdating":false,"isAdopting":false,"isAdopted":true,"isAdoptedByOther":false,"isProvisioned":true,"isRebooting":false,"isSshEnabled":false,"canAdopt":false,"isAttemptingToConnect":false,"lastMotion":1638891279630,"micVolume":100,"isMicEnabled":true,"isRecording":true,"isMotionDetected":true,"isSmartDetected":false,"phyRate":28,"hdrMode":true,"videoMode":"default","isProbingForWifi":false,"apMac":null,"apRssi":null,"elementInfo":null,"chimeDuration":0,"isDark":true,"lastPrivacyZonePositionId":null,"lastRing":null,"isLiveHeatmapEnabled":false,"anonymousDeviceId":"84ac9459-648f-5f89-ae1d-4f68e4e52629","eventStats":{"motion":{"today":2926,"average":88,"lastDays":[0,616],"recentHours":[144]},"smart":{"today":0,"average":0,"lastDays":[0,0]}},"videoReconfigurationInProgress":false,"voltage":22.5,"wiredConnectionState":{"phyRate":null},"channels":[{"id":0,"videoId":"video1","name":"High","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":1600,"height":1200,"fps":30,"bitrate":6000000,"minBitrate":32000,"maxBitrate":6000000,"minClientAdaptiveBitRate":0,"minMotionAdaptiveBitRate":750000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25,30],"idrInterval":5},{"id":1,"videoId":"video3","name":"Medium","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":960,"height":720,"fps":30,"bitrate":1200000,"minBitrate":32000,"maxBitrate":2000000,"minClientAdaptiveBitRate":150000,"minMotionAdaptiveBitRate":750000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25,30],"idrInterval":5},{"id":2,"videoId":"video2","name":"Low","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":480,"height":360,"fps":30,"bitrate":200000,"minBitrate":32000,"maxBitrate":1000000,"minClientAdaptiveBitRate":0,"minMotionAdaptiveBitRate":200000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25,30],"idrInterval":5}],"ispSettings":{"aeMode":"auto","irLedMode":"auto","irLedLevel":255,"wdr":1,"icrSensitivity":0,"brightness":50,"contrast":50,"hue":50,"saturation":50,"sharpness":50,"denoise":50,"isFlippedVertical":false,"isFlippedHorizontal":false,"isAutoRotateEnabled":false,"isLdcEnabled":true,"is3dnrEnabled":true,"isExternalIrEnabled":false,"isAggressiveAntiFlickerEnabled":false,"isPauseMotionEnabled":false,"dZoomCenterX":50,"dZoomCenterY":50,"dZoomScale":0,"dZoomStreamId":4,"focusMode":"ztrig","focusPosition":0,"touchFocusX":1001,"touchFocusY":1001,"zoomPosition":0,"mountPosition":null},"talkbackSettings":{"typeFmt":"aac","typeIn":"serverudp","bindAddr":"0.0.0.0","bindPort":7004,"filterAddr":"","filterPort":0,"channels":1,"samplingRate":22050,"bitsPerSample":16,"quality":100},"osdSettings":{"isNameEnabled":true,"isDateEnabled":true,"isLogoEnabled":true,"isDebugEnabled":false},"ledSettings":{"isEnabled":false,"blinkRate":0},"speakerSettings":{"isEnabled":true,"areSystemSoundsEnabled":false,"volume":80},"recordingSettings":{"prePaddingSecs":10,"postPaddingSecs":10,"minMotionEventTrigger":500,"endMotionEventDelay":3000,"suppressIlluminationSurge":false,"mode":"always","geofencing":"off","motionAlgorithm":"enhanced","enablePirTimelapse":false,"useNewMotionAlgorithm":true},"smartDetectSettings":{"objectTypes":["person","vehicle"]},"recordingSchedules":[],"motionZones":[{"id":2,"name":"New Zone","color":"#AB46BC","points":[[0,0],[0.99975586,0],[0.99975586,1],[0.0014648438,0.99987704]],"sensitivity":22}],"privacyZones":[{"id":1,"name":"New Zone","color":"#586CED","points":[[0,0.015],[0.237,0.128],[0.393,0.419],[0.362,0.486],[0.356,0.441],[0.025,0.439],[0.024,0.64],[0,0.646]]}],"smartDetectZones":[{"id":1,"name":"New Zone","color":"#AB46BC","points":[[0.487,0.445],[0.483,0.25],[1,0],[1,1],[0.492,1],[0,0.988],[0,0.674]],"sensitivity":100,"objectTypes":["person","vehicle"]}],"smartDetectLines":[],"stats":{"rxBytes":236330256,"txBytes":9222328007,"wifi":{"channel":5,"frequency":2432,"linkSpeedMbps":null,"signalQuality":100,"signalStrength":-65},"battery":{"percentage":null,"isCharging":false,"sleepState":"disconnected"},"video":{"recordingStart":1638779373627,"recordingEnd":1638891278690,"recordingStartLQ":1638627110284,"recordingEndLQ":1638891278686,"timelapseStart":1638627110245,"timelapseEnd":1638891138659,"timelapseStartLQ":1638627110245,"timelapseEndLQ":1638891148653},"storage":{"used":42949672960,"rate":772.348712249141},"wifiQuality":100,"wifiStrength":-65},"featureFlags":{"canAdjustIrLedLevel":false,"canMagicZoom":false,"canOpticalZoom":false,"canTouchFocus":false,"hasAccelerometer":false,"hasAec":true,"hasBattery":false,"hasBluetooth":true,"hasChime":true,"hasExternalIr":false,"hasIcrSensitivity":true,"hasLdc":true,"hasLedIr":true,"hasLedStatus":true,"hasLineIn":false,"hasMic":true,"hasPrivacyMask":true,"hasRtc":false,"hasSdCard":false,"hasSpeaker":true,"hasWifi":true,"hasHdr":true,"hasAutoICROnly":true,"videoModes":["default","highFps"],"videoModeMaxFps":[24,45],"hasMotionZones":true,"hasLcdScreen":true,"mountPositions":[],"smartDetectTypes":["person","vehicle"],"motionAlgorithms":["enhanced"],"hasSquareEventThumbnail":true,"hasPackageCamera":false,"privacyMaskCapability":{"maxMasks":16,"rectangleOnly":false},"focus":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"pan":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"tilt":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"zoom":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"hasSmartDetect":true},"pirSettings":{"pirSensitivity":100,"pirMotionClipLength":15,"timelapseFrameInterval":15,"timelapseTransferInterval":600},"lcdMessage":{"type":"CUSTOM_MESSAGE","text":"Kamenzer Strasse 17-19","resetAt":null},"wifiConnectionState":{"channel":5,"frequency":2432,"phyRate":28,"signalQuality":100,"signalStrength":-65,"ssid":"SIO"},"lenses":[],"id":"61ab76000174a5038700085b","isConnected":true,"platform":"s5l","hasSpeaker":true,"hasWifi":true,"audioBitrate":64000,"canManage":false,"isManaged":true,"modelKey":"camera"}`

**Comment by AngellusMortis** [comment]
Huh. Yeah. I have _no_ idea why, but UniFi Protect is tell me that your doorbell has never been rung:

![image](https://user-images.githubusercontent.com/490848/145059545-2be0d4c0-4a50-49d1-b094-7c145e301eb9.png)

**Comment by AngellusMortis** [comment]
Also, can you go to `/proxy/protect/api/events?type=ring&limit=100`? Just tell me if you see the ring events there. I do not need any of the JSON from it.

Expected output if you are seeing ring events: 
![image](https://user-images.githubusercontent.com/490848/145060000-e1667efa-11a4-4d1f-9cfb-781bbc5ec79f.png)

**Comment by AngellusMortis** [comment]
AH. So, you have 3 doorbell and one or more have them have _not_ been rung. Okay. That is the part I was missing. in that case, as a work around, go around to each doorbell and ring them at least once. That should resolve everything for you. 

But I will leave this ticket open until I make the necessary code changes to support doorbells that have never been rung.

**Comment by RW-7** [comment]
there are some
`[{"id":"616d5b95018ee90387003dfa","type":"ring","start":1634556821397,"end":1634556822397,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-616d5b95018ee90387003dfa","heatmap":"e-616d5b95018ee90387003dfa","modelKey":"event"},{"id":"616e89cb0230e90387004819","type":"ring","start":1634634187557,"end":1634634188557,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-616e89cb0230e90387004819","heatmap":"e-616e89cb0230e90387004819","modelKey":"event"},{"id":"616ea41e00c8e90387004927","type":"ring","start":1634640926199,"end":1634640927199,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-616ea41e00c8e90387004927","heatmap":"e-616ea41e00c8e90387004927","modelKey":"event"},{"id":"617033b20189e903870056ea","type":"ring","start":1634743218392,"end":1634743219392,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617033b20189e903870056ea","heatmap":"e-617033b20189e903870056ea","modelKey":"event"},{"id":"617132be0126e903870060c2","type":"ring","start":1634808510293,"end":1634808511293,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617132be0126e903870060c2","heatmap":"e-617132be0126e903870060c2","modelKey":"event"},{"id":"617135ed01bce90387006122","type":"ring","start":1634809325442,"end":1634809326442,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617135ed01bce90387006122","heatmap":"e-617135ed01bce90387006122","modelKey":"event"},{"id":"61714e0e01fae903870062f6","type":"ring","start":1634815502505,"end":1634815503505,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61714e0e01fae903870062f6","heatmap":"e-61714e0e01fae903870062f6","modelKey":"event"},{"id":"617690d90044d70387003654","type":"ring","start":1635160281066,"end":1635160282066,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617690d90044d70387003654","heatmap":"e-617690d90044d70387003654","modelKey":"event"},{"id":"617690f701e8d70387003661","type":"ring","start":1635160311486,"end":1635160312486,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617690f701e8d70387003661","heatmap":"e-617690f701e8d70387003661","modelKey":"event"},{"id":"61769e770212d70387003828","type":"ring","start":1635163767529,"end":1635163768529,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61769e770212d70387003828","heatmap":"e-61769e770212d70387003828","modelKey":"event"},{"id":"6176cb5401f7d70387003acd","type":"ring","start":1635175252501,"end":1635175253501,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-6176cb5401f7d70387003acd","heatmap":"e-6176cb5401f7d70387003acd","modelKey":"event"},{"id":"6177c360000dd703870045b9","type":"ring","start":1635238752012,"end":1635238753012,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-6177c360000dd703870045b9","heatmap":"e-6177c360000dd703870045b9","modelKey":"event"},{"id":"6177d46f02add70387004657","type":"ring","start":1635243119681,"end":1635243120681,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-6177d46f02add70387004657","heatmap":"e-6177d46f02add70387004657","modelKey":"event"},{"id":"617a6e4a015cd7038700655c","type":"ring","start":1635413578344,"end":1635413579344,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617a6e4a015cd7038700655c","heatmap":"e-617a6e4a015cd7038700655c","modelKey":"event"},{"id":"617a884601fad703870066f3","type":"ring","start":1635420230505,"end":1635420231505,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617a884601fad703870066f3","heatmap":"e-617a884601fad703870066f3","modelKey":"event"},{"id":"617ec34f0152d70387009597","type":"ring","start":1635697487337,"end":1635697488337,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617ec34f0152d70387009597","heatmap":"e-617ec34f0152d70387009597","modelKey":"event"},{"id":"617ed60801aa0f038700055f","type":"ring","start":1635702280425,"end":1635702281425,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617ed60801aa0f038700055f","heatmap":"e-617ed60801aa0f038700055f","modelKey":"event"},{"id":"617edd1a007a0f03870005d1","type":"ring","start":1635704090121,"end":1635704091121,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617edd1a007a0f03870005d1","heatmap":"e-617edd1a007a0f03870005d1","modelKey":"event"},{"id":"617fc42501b8670387000819","type":"ring","start":1635763237390,"end":1635763238390,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-617fc42501b8670387000819","heatmap":"e-617fc42501b8670387000819","modelKey":"event"},{"id":"6181272100387403870004a5","type":"ring","start":1635854113054,"end":1635854114054,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-6181272100387403870004a5","heatmap":"e-6181272100387403870004a5","modelKey":"event"},{"id":"618e80340084a80387008367","type":"ring","start":1636728884119,"end":1636728885119,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-618e80340084a80387008367","heatmap":"e-618e80340084a80387008367","modelKey":"event"},{"id":"6190be10034d6f0387001b27","type":"ring","start":1636875792844,"end":1636875793844,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-6190be10034d6f0387001b27","heatmap":"e-6190be10034d6f0387001b27","modelKey":"event"},{"id":"619f64ee0281d2038700112a","type":"ring","start":1637836014640,"end":1637836015640,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-619f64ee0281d2038700112a","heatmap":"e-619f64ee0281d2038700112a","modelKey":"event"},{"id":"619f81b00101d2038700129b","type":"ring","start":1637843376255,"end":1637843377255,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-619f81b00101d2038700129b","heatmap":"e-619f81b00101d2038700129b","modelKey":"event"},{"id":"619f82710290d203870012c9","type":"ring","start":1637843569654,"end":1637843570654,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-619f82710290d203870012c9","heatmap":"e-619f82710290d203870012c9","modelKey":"event"},{"id":"61a0b80f03b1d20387002465","type":"ring","start":1637922831943,"end":1637922832943,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61a0b80f03b1d20387002465","heatmap":"e-61a0b80f03b1d20387002465","modelKey":"event"},{"id":"61a4c0b501afd20387005484","type":"ring","start":1638187189430,"end":1638187190430,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61a4c0b501afd20387005484","heatmap":"e-61a4c0b501afd20387005484","modelKey":"event"},{"id":"61a5f9b201b8d20387006468","type":"ring","start":1638267314439,"end":1638267315439,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61a5f9b201b8d20387006468","heatmap":"e-61a5f9b201b8d20387006468","modelKey":"event"},{"id":"61a610ba00ecd20387006530","type":"ring","start":1638273210233,"end":1638273211233,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61a610ba00ecd20387006530","heatmap":"e-61a610ba00ecd20387006530","modelKey":"event"},{"id":"61ab918500b0a50387000b5e","type":"ring","start":1638633861175,"end":1638633862175,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"61ab76000174a5038700085b","partition":null,"user":null,"metadata":{},"thumbnail":"e-61ab918500b0a50387000b5e","heatmap":"e-61ab918500b0a50387000b5e","modelKey":"event"},{"id":"61acf8aa02afa30387000db8","type":"ring","start":1638725802686,"end":1638725803686,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"61ab76000174a5038700085b","partition":null,"user":null,"metadata":{},"thumbnail":"e-61acf8aa02afa30387000db8","heatmap":"e-61acf8aa02afa30387000db8","modelKey":"event"},{"id":"61add997012ba303870021ab","type":"ring","start":1638783383287,"end":1638783384287,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"61ab76000174a5038700085b","partition":null,"user":null,"metadata":{},"thumbnail":"e-61add997012ba303870021ab","heatmap":"e-61add997012ba303870021ab","modelKey":"event"},{"id":"61af344e01f0a30387006151","type":"ring","start":1638872142491,"end":1638872143491,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"61addbd5007fa303870021e6","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af344e01f0a30387006151","heatmap":"e-61af344e01f0a30387006151","modelKey":"event"},{"id":"61af388f01aca303870063e1","type":"ring","start":1638873231427,"end":1638873232427,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af388f01aca303870063e1","heatmap":"e-61af388f01aca303870063e1","modelKey":"event"},{"id":"61af39170274a30387006424","type":"ring","start":1638873367626,"end":1638873368626,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af39170274a30387006424","heatmap":"e-61af39170274a30387006424","modelKey":"event"},{"id":"61af394701b5a3038700643c","type":"ring","start":1638873415435,"end":1638873416435,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af394701b5a3038700643c","heatmap":"e-61af394701b5a3038700643c","modelKey":"event"},{"id":"61af53730121a30387007339","type":"ring","start":1638880115280,"end":1638880116280,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af53730121a30387007339","heatmap":"e-61af53730121a30387007339","modelKey":"event"},{"id":"61af703300a6a30387008410","type":"ring","start":1638887475165,"end":1638887476165,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af703300a6a30387008410","heatmap":"e-61af703300a6a30387008410","modelKey":"event"},{"id":"61af771d03bca30387008801","type":"ring","start":1638889245955,"end":1638889246955,"score":0,"smartDetectTypes":[],"smartDetectEvents":[],"camera":"6103df73026d730387003325","partition":null,"user":null,"metadata":{},"thumbnail":"e-61af771d03bca30387008801","heatmap":"e-61af771d03bca30387008801","modelKey":"event"}]`

**Comment by AngellusMortis** [comment]
I went ahead and deleted your JSON since I got the data I needed from them. Let me know if ringing the doorbells gets everything working for you now. Otherwise I can prioritize trying to get the fix for the lastRing = null fix sooner.

**Comment by RW-7** [comment]
yes now it works again, thanks..
where can I find unifiprotect.add_doorbell_text (is nicer in HA than via the APP)

**Comment by AngellusMortis** [comment]
On the left side bar of Home Assistant -> Developer Tools -> Services. URL should be: `/developer-tools/service`. Then search for "Doorbell".

![image](https://user-images.githubusercontent.com/490848/145063309-97ab6f3a-f5cc-4219-a8f2-0d9c6def874b.png)

Just note that you will need to restart HA after adding all of your custom doorbell messages for them to appear in the Select entity. This is a limitation of how downstream integrations ingest options for Select entities so they cannot be dynamically updated.

**Comment by RW-7** [comment]
ok, great now everything works as usual,
thank you for your help, and of course for creating and maintaining your project

**Comment by AngellusMortis** [comment]
The underlaying issue should be fully resolved in 0.11-beta3 now.
